roaring/roaring_internal_test.go

```diff
@@ -3258,38 +3258,81 @@ func runContainerFunc(f interface{}, c ...*Container) *Container {
 	return nil
 }
 
-func TestUnmarshalOfficialRoaring(t *testing.T) {
-	//generated serialize image from java(clojure) with arrays
-	rbContainerWithTwoArrays, _ := hex.DecodeString("3A300000020000000000020001000000180000001E0000000100020003000100")
-	bm := NewBitmap()
-	er := bm.UnmarshalBinary(rbContainerWithTwoArrays)
-	if er != nil {
-		t.Fatalf("UnmarshalOfficialRoaring %s", er)
-	}
-	if bm.Count() != 4 {
-		t.Fatalf("unexpected bitmap %v expected bits [1 2 3 65537]", bm.Slice())
-	}
-	//generated serialize image from java(clojure) with a run and array
-	rbContainerWithRLEandArray, _ := hex.DecodeString("3B3001000100000900010000000100010009000100")
-	bm = NewBitmap()
-	er = bm.UnmarshalBinary(rbContainerWithRLEandArray)
-	if er != nil {
-		t.Fatalf("UnmarshalOfficialRoaring %s", er)
-	}
-	if bm.Count() != 11 {
-		t.Fatalf("unexpected bitmap %v expected bits [1 2 3 4 5 6 7 8 9 10 65537]", bm.Slice())
-	}
-	//had to use an external file because emacs was barfing on the long line :()
-	_bitmap_array_container, _ := ioutil.ReadFile("testdata/bitmapcontainer.roaringbitmap")
-	bm = NewBitmap()
-	er = bm.UnmarshalBinary(_bitmap_array_container)
-	if er != nil {
-		t.Fatalf("UnmarshalOfficialRoaring %s", er)
-	}
-	if bm.Count() != 10000 {
-		t.Fatalf("expecting X got %d", bm.Count())
+func TestUnmarshalRoaringWithNoErrors(t *testing.T) {
+	testValues := []struct {
+		hexString string
+		count uint64
+		expectedBits string
+	}{
+		{	// generated serialize image from java(clojure) with arrays
+			hexString : "3A300000020000000000020001000000180000001E0000000100020003000100",
+			count : 4,
+			expectedBits : "[1 2 3 65537]",
+		},
+		{	// generated serialize image from java(clojure) with a run and array
+			hexString : "3B3001000100000900010000000100010009000100",
+			count : 11,
+			expectedBits : "[1 2 3 4 5 6 7 8 9 10 65537]",
+		},
+		{	// had to use an external file because emacs was barfing on the long line :()
+			hexString : "testdata/bitmapcontainer.roaringbitmap",
+			count : 10000,
+			expectedBits : "X",
+		},
+
+	}
+	for _, testLoop := range testValues {
+		if testLoop.hexString != "testdata/bitmapcontainer.roaringbitmap" {
+			testContainer, _ := hex.DecodeString(testLoop.hexString)
```

Comment:
The two sections of this if-else block look pretty similar - I think some of the code could be combined (and moved after the if-else), to both be more readable and avoid bugs when editing one section but not the other.